src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.Fixer.cs

```diff
@@ -145,7 +150,11 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
             }
             SyntaxNode newInvocationExpression = generator.InvocationExpression(asyncMethodNode, nodeArguments).WithTriviaFrom(streamInstanceNode);
 
-            SyntaxNode newRoot = generator.ReplaceNode(root, invocation.Syntax, newInvocationExpression.WithTriviaFrom(invocation.Syntax));
+            bool containsSystemImport = IsSystemNamespaceImported(generator.GetNamespaceImports(root));
+
+            SyntaxNode rootWithImports = containsSystemImport ? root : generator.AddNamespaceImports(root, generator.NamespaceImportDeclaration("System").WithAddImportsAnnotation());
```

Comment:
Is `WithAddImportAnnotation` necessary here? I was trying to see if that would solve the unit test failure, and decided to leave it so I could ask here in the review. Should I remove it?

Reply:
I was thinking the other way. `WithAddImportAnnotation` should be the supported way to add an imports instead of explicitly requiring to add a namespace import using the generator. I think @sharwell is the better person to answer this question.

Reply:
> Is WithAddImportAnnotation necessary here?

No, it doesn't make sense to use this annotation on using directives.